branchsearch.cpp

```diff
@@ -93,6 +93,40 @@ void filter_branches(vector<string> &all_branches, char* regex_value, vector<str
     }
 }
 
+bool best_match(vector<string> &all_branches, char* pattern, string& res){
+    // pattern to lowercase
+    std::transform(pattern, pattern+strlen(pattern), pattern, [](unsigned char c){ return std::tolower(c); });
+
+    // take the ones containing the pattern
+    vector<string*> matches;
+    for (string &b: all_branches){
+        // branch to lowercase
+        std::transform(b.begin(), b.end(), b.begin(), [](unsigned char c){ return std::tolower(c); });
```

Comment:
This transformation is a problem, because you will change the branch name and you return that as well. You should keep the original one. I know because I tried it and it broke:
```
➜  git:(staging) ✗ bs GZ-66
error: pathspec 'gz-66-if' did not match any file(s) known to git.
```

Some tests would have prevented this potential regression :smile:

Reply:
Ah right, will fix that. 
(Feel free to send a PR with some tests)

Reply:
There you go, check it out now